apps/common/models/pdf.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+const Model = require('hof-model');
+const config = require('../../../config');
+
+module.exports = class PDFModel extends Model {
+
+  url() {
+    return config.pdf.url;
+  }
+
+  handleResponse(response, callback) {
+    try {
+      JSON.parse(response.body || '{}');
+      let err = new Error();
+      err.body = response.body;
+      err.status = response.statusCode;
+      return callback(err, null, response.statusCode);
+    } catch (err) {
+      this.parseResponse(response.statusCode, response.body, callback);
+    }
+  }
```

Comment:
I don't really get this function. It seems to throw an error when there isn't an error, and not throw an error when there is.

What is the intention?

---

Edit: I *think* I get it now. If it's a PDF (as we expect) then `JSON.parse` will throw and we end up in the catch block. Whereas errors will respond with JSON, so we know that if we have JSON then an error has occurred. I'm not sure I like it though. It's super confusing to have the success/error states apparently backwards like that.